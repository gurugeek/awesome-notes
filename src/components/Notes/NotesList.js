import React, { PureComponent } from 'react';
import { View, Text, VirtualizedList } from 'react-native';
import PropTypes from 'prop-types';

class NotesList extends PureComponent {

    _renderItem({ item }) {
        return (
            <View style={{ height: 100, width: '100%', alignItems: 'center', justifyContent: 'center' }}>
                <Text>{item.text}</Text>
            </View>
        );
    }

    _getItem(data, index) {
        return data[index];
    }

    _shouldItemUpdate(prev, next) {
        return prev.item !== next.item;
    }

    render() {
        console.log(this.props.notes);
        return (
            <View style={{ flex: 1 }}>
                <Text>Notes List</Text>
                <VirtualizedList
                    data={this.props.notes}
                    renderItem={this._renderItem}
                    shouldItemUpdate={this._shouldItemUpdate}
                    windowSize={10}
                    keyExtractor={(item, index) => item.id}
                    getItemCount={(data) => data.length}
                    getItem={this._getItem}
                    showsVerticalScrollIndicator={false}
                />
            </View>
        );
    }
}

NotesList.propTypes = {
    notes: PropTypes.array.isRequired
};

export default NotesList;