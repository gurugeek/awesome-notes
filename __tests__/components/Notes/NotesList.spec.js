import React from 'react';
import { View, VirtualizedList } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import NotesList from '../../../src/components/Notes/NotesList';

const notes = [{ id: 'awesomeId', text: 'Awesome text', favourite: false, createdAt: new Date(601254000000)}];

describe('<NotesList /> Component', () => {
    it('NotesList should match with snapshot', () => {
        const tree = renderer.create(
            <NotesList notes={notes} updateNote={() => {}} deleteNote={() => {}}/>
        );
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const notesListShallow = shallow(<NotesList notes={notes} updateNote={() => {}} deleteNote={() => {}}/>);

    it('NotesList should exists', () => {
        expect(notesListShallow.length).toEqual(1);
    });

    it('NotesList should has one View', () => {
        expect(notesListShallow.find(View)).toHaveLength(1);
    });

    it('NotesList should has one VirtualizedList', () => {
        expect(notesListShallow.find(VirtualizedList)).toHaveLength(1);
    });
});