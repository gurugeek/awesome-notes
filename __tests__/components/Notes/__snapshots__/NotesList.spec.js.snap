// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotesList /> Component NotesList should match with snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "createdAt": 1989-01-19T23:00:00.000Z,
          "favourite": false,
          "id": "awesomeId",
          "text": "Awesome text",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    shouldItemUpdate={[Function]}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={10}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#dbddde",
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "transparent",
                "marginBottom": 10,
              },
            ]
          }
        >
          <View
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              Array [
                Object {},
                Object {
                  "left": 0,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "backgroundColor": "#A8DADC",
                  "borderRadius": 10,
                  "opacity": 1,
                  "padding": 10,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "perspective": 1000,
                    },
                    Object {
                      "rotateY": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "marginBottom": 5,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#211A1E",
                      "fontFamily": "Roboto-Regular",
                    }
                  }
                >
                  Awesome text
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#457B9D",
                      "fontFamily": "Roboto-Thin",
                      "fontSize": 10,
                      "textAlign": "right",
                    }
                  }
                >
                  20/1/1989
                </Text>
              </View>
            </View>
          </View>
          <View />
          <View />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
